mender-update/daemon/states.cpp

```diff
@@ -401,13 +407,33 @@ void SendStatusUpdateState::DoStatusUpdate(Context &ctx, sm::EventPoster<StateEv
 		}
 	}
 
+	// Push status.
 	auto err = ctx.deployment_client->PushStatus(
 		ctx.deployment.state_data->update_info.id,
 		status,
 		"",
 		ctx.mender_context.GetConfig().server_url,
 		ctx.http_client,
-		result_handler);
+		[result_handler, &ctx](error::Error err) {
+			// If the deployment succeeded, we don't submit logs, so there's nothing
+			// more to do.
+			if (err != error::NoError || !ctx.deployment.failed) {
+				result_handler(err);
+				return;
+			}
+
+			// Push logs.
+			err = ctx.deployment_client->PushLogs(
+				ctx.deployment.state_data->update_info.id,
+				ctx.deployment.logger->LogFilePath(),
+				ctx.mender_context.GetConfig().server_url,
+				ctx.http_client,
+				result_handler);
```

Comment:
Pushing the logs will be tried only once? (Probably covered somewhere else, I am new to the state machine)

Reply:
The `result_handler` has retry logic inside it, so it's initiated from there.